halo2_proofs/src/poly/evaluator.rs

```diff
@@ -296,13 +328,24 @@ pub(crate) enum Ast<E, F: Field, B: Basis> {
     ConstantTerm(F),
 }
 
+impl<E, F: Field, B: Basis> Ast<E, F, B> {
+    pub fn distribute_powers<I: IntoIterator<Item = Self>>(i: I, base: F) -> Self {
+        Ast::DistributePowers(Arc::new(i.into_iter().collect()), base)
```

Comment:
At some point I might try and see if I can find a way to avoid this allocation, but it's going to require a bunch of extra lifetime annotations, so only worthwhile if we actually would see a performance benefit.